projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.pipes.ts

```diff
@@ -82,7 +82,9 @@ export class IgxPivotAutoTransform implements PipeTransform {
     protected processCollectionToPivotRecord(config: IPivotConfiguration, collection: any[]): IPivotGridRecord[] {
         const pivotKeys: IPivotKeys = config.pivotKeys || DEFAULT_PIVOT_KEYS;
         const enabledRows = config.rows.filter(x => x.enabled);
-        const allFlat: IPivotDimension[] = PivotUtil.flatten(enabledRows);
+        const allFlat: IPivotDimension[] = cloneArray(
+            PivotUtil.flatten(enabledRows), true, new Set<string>(["externalObject"]));
```

Comment:
Oh, um, yeah... can't say I'm a huge fan of baking in an external 'magic' property name in the source.

Reply:
The more I look at this, the more I'm thinking we should consider a pivot (tehe) to some sort of clone strategy that can be overridden to serve per-case needs. I seem to recall there already is something like that added, though obviously not used for this clone.

Reply:
In any case, the above suggestion might end up taking more time and API adjustments, so hey - might have to be okay with this for now and just plan on replacing it with a strategy down the line